imap/append.c

```diff
@@ -1257,7 +1257,8 @@ HIDDEN int append_run_annotator(struct appendstate *as,
 
     /* Reset user flags */
     uint32_t user_flags[MAX_USER_FLAGS/32];
-    memset(user_flags, 0, sizeof(user_flags)/sizeof(user_flags[0]));
```

Comment:
Both the original and the fixed version seem bogus here.  It will only zero out the first quarter of the array?

The `sizeof(foo) / sizeof(foo[0])` idiom is for calculating the number of array elements.  But memset's third argument is a size in bytes, not a number of elements.

It should be correct to do `memset(user_flags, 0, sizeof(user_flags));` (and, this should also stop gcc7 from whinging about it).